### PR TITLE
Multi stage pipeline

### PR DESCRIPTION
**Pull request by vincentmeier**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -2,90 +2,147 @@ trigger:
...
+              Platform: 'Any CPU'
+              BuildConfiguration: 'Release'
+        workspace:
+          clean: outputs
+
+        steps:
+          - checkout: self
+            clean: true
+
+          - task: GitVersion@5
```
Comment (Philipp-Navis) [inline azure-pipelines.yml:26]: Reference to `GitVersion.yml` missing

File: azure-pipelines.yml [hunk 2]
```diff
@@ -2,90 +2,147 @@ trigger:
...
+              restoreSolution: BlueTracker.SDK.sln
+
+          - task: VSBuild@1
+            displayName: 'Build solution BlueTracker.SDK.sln'
+            inputs:
+              solution: BlueTracker.SDK.sln
+              vsVersion: '16.0'
+              platform: '$(Platform)'
+              configuration: '$(BuildConfiguration)'
+              clean: true
```
Comment (Philipp-Navis) [inline azure-pipelines.yml:48]: `clean: true` is not needed as we clean the repository at the beginning of the job.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -2,90 +2,147 @@ trigger:
...
+            inputs:
+              solution: BlueTracker.SDK.sln
+              msbuildVersion: '16.0'
+              platform: '$(Platform)'
+              configuration: '$(BuildConfiguration)'
+              msbuildArguments: '/t:pack /p:PackageOutputPath="$(Build.ArtifactStagingDirectory)" /p:NoPackageAnalysis=true /p:PackageVersion=$(GitVersion.FullSemVer)'
+              maximumCpuCount: true
+
+          # Trigger when called from build pipeline, NOT master
+          - task: NuGetCommand@2
```
Comment (Philipp-Navis) [inline azure-pipelines.yml:61]: This step is not needed as it's performed separately in the deployment job

File: azure-pipelines.yml [hunk 4]
```diff
@@ -2,90 +2,147 @@ trigger:
...
+            displayName: 'NuGet push'
+            inputs:
+              command: push
+              packagesToPush: '$(Build.ArtifactStagingDirectory)/*.nupkg'
+              publishVstsFeed: 'c69ffa3a-dc58-431c-afe0-12f465fcb265'
+            condition: and(succeeded(), ne(variables['Build.sourceBranch'], 'refs/heads/master'))
+
+          - task: CopyFiles@2
```
Comment (Philipp-Navis) [inline azure-pipelines.yml:69]: The only artifact that is needed in the later steps is the nuget package. This and the following two tasks can be simplified.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -2,90 +2,147 @@ trigger:
...
+          # Trigger when called from build pipeline, NOT master
+          - task: PublishBuildArtifacts@1
+            displayName: 'Any Branch: Publish Artifact'
+            inputs:
+              PathtoPublish: '$(build.artifactstagingdirectory)'
+            condition: and(succeededOrFailed(), ne(variables['Build.sourceBranch'],'refs/heads/master'))
+
+  - stage: DeployToAzure
+    displayName: 'Deploy to Azure DevOps'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
```
Comment (Philipp-Navis) [inline azure-pipelines.yml:96]: To align with the previous functionality, the deployment to azure should always be running.

Comment (vincentmeier) [reply azure-pipelines.yml]: Removed in: d1c79bc

File: azure-pipelines.yml (hunk 3 above)
Comment (vincentmeier) [reply azure-pipelines.yml]: Resolved in: ca126f3

File: azure-pipelines.yml (hunk 2 above)
Comment (vincentmeier) [reply azure-pipelines.yml]: Resolved in: 2760405

File: azure-pipelines.yml (hunk 1 above)
Comment (vincentmeier) [reply azure-pipelines.yml:33]: Resolved in: f5bdbca

File: azure-pipelines.yml (hunk 4 above)
Comment (Philipp-Navis) [reply azure-pipelines.yml:67]: This `copy files` step is no longer needed as the `NuGet package` step already publishes the package to the `ArtifactsStagingDirectory`.

Comment (vincentmeier) [reply azure-pipelines.yml]: Solved in: 5d52447

**Pull request merged**